features/step_definitions/sample_file_steps.rb

```diff
@@ -0,0 +1,219 @@
+Given(/^a demo directory with a smelly file$/) do
+  contents = <<-EOS
+class Dirty
+  # This method smells of :reek:NestedIterators but ignores them
+  def awful(x, y, offset = 0, log = false)
+    puts @screen.title
+    @screen = widgets.map {|w| w.each {|key| key += 3}}
+    puts @screen.contents
+  end
+end
+  EOS
+  write_file('demo/demo.rb', contents)
+end
+
+Given(/^a smelly file called 'smelly.rb'$/) do
+  contents = <<-EOS
+# smelly class for testing purposes
+class Dirty
```

Comment:
Almost identical to above, can we merge this maybe? (And pleaaaase get rid of "fred"?)